.github/workflows/release.yaml

```diff
@@ -0,0 +1,90 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Extract version from tag
+        id: extract_version
+        run: echo "VERSION=${GITHUB_REF#refs/tags/v}" >> $GITHUB_ENV
+
+      - name: Debug version
+        run: echo ${{ env.VERSION }}
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login Helm to DockerHub registry
+        run: |
+           helm registry login registry-1.docker.io -u ${{ secrets.DOCKERHUB_USERNAME }} --password-stdin <<< "${{ secrets.DOCKERHUB_TOKEN }}"
+
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Build and publish Docker image (init container)
+        env:
+          VERSION: ${{ env.VERSION }}
+        run: |
+          echo "Building Docker init image with version: ${VERSION}"
+          docker build -t kubeshop/monokle-admission-controller-init:latest -t kubeshop/monokle-admission-controller-init:${VERSION} ./admission-controller/init/
+          docker push kubeshop/monokle-admission-controller-init:${VERSION}
+          docker push kubeshop/monokle-admission-controller-init:latest
+
+      - name: Build and publish Docker image (server container)
+        env:
+          VERSION: ${{ env.VERSION }}
+        run: |
+          echo "Building Docker server image with version: ${VERSION}"
+          docker build -t kubeshop/monokle-admission-controller-server:latest -t kubeshop/monokle-admission-controller-server:${VERSION} ./admission-controller/server/
+          docker push kubeshop/monokle-admission-controller-server:${VERSION}
+          docker push kubeshop/monokle-admission-controller-server:latest
```

Comment:
Why not use the docker GH actions that are designed for this exact purpose of building and pushing a docker image? They also bring the benefit of layer caching, which reduces the amount of time we spend building.

Furthermore, why not build these docker images in separate actions? 
I think there's a benefit in that by not building the initContainer image each time as that will probably not change as often as the main image.

Reply:
> Why not use the docker GH actions that are designed for this exact purpose of building and pushing a docker image? They also bring the benefit of layer caching, which reduces the amount of time we spend building.

Agree, I'll be happy to improve it. I just went with quite basic version to not spend too much time on it. Do you mean to use something like buildx (https://docs.docker.com/build/cache/backends/gha/), also similarly we have here - https://github.com/kubeshop/monokle-saas/blob/e3efa7c43904aac4f64bd1f4b27abe99dbd7a6d5/.github/workflows/api-renderers.yml#L41-L61?

> Furthermore, why not build these docker images in separate actions?
I think there's a benefit in that by not building the initContainer image each time as that will probably not change as often as the main image.

The idea was to have separate image version for each release, but I guess it can be done just by tagging already built/existing images if it's source was not modified? (not sure, about Helm in DockerHub OCI registry, haven't found a way to add multiple tags to single version 🤔).

Reply:
I can somehow see why we would want the same version on both, but on the other hand it does not necessarily make sense to version the initContainer if it does not actually change. IDK, @WitoDelnat, what do you think?

Reply:
I'd keep the versions separated. There is little benefit to having them being the same. If you have problems with certificates it can even help to see if the init container changed (meaningfully).

As for the Helm Chart versioning I'm a bit more hesitant. Do we want to publish a new Chart for each Docker Image or leave the image as a variable in the values.yaml and default it to `latest`. @murarustefaan you have knowledge about current best practices?

Reply:
Sure. TBH I have not seen a lot of projects linking the Chart and App Versions, they are separated for a reason. 
Chart version should change only on Chart modifications, while 
App Version should change alongside the image version. Here I would link the App Version to the admission controller version (which should be configurable trough values, just in case) while leaving the init container set to latest.

Reply:
After discussing further with @murarustefaan I extracted an issue, since it seems to be broad topic and it's difficult to know what will be really useful at this stage - https://github.com/kubeshop/monokle-admission-controller/issues/13.

For improvements:
- Changed to multi-stage images - https://github.com/kubeshop/monokle-admission-controller/pull/9/commits/e5eb0f4baff4d86877cab4e476451e1b4b95ce6e
- Utilized buildx action for docker related logic - https://github.com/kubeshop/monokle-admission-controller/pull/9/commits/e98d709105ca94914fdd8fc8ed5a7561381bebfd
- Added build cache and restructured job - https://github.com/kubeshop/monokle-admission-controller/pull/9/commits/1802bcd0947c60d0c0646d54cfe294298f60f959